src/sentry/api/endpoints/project_event_details.py

```diff
@@ -41,19 +42,52 @@ def get(self, request, project, event_id):
         :auth: required
         """
 
+        use_snuba = request.GET.get('enable_snuba') == '1' and \
+            options.get('snuba.events-queries.enabled')
+
+        if not use_snuba:
+            return self.get_legacy(request, project, event_id)
+
+        # We might get a database id instead of an event id
+        if not is_event_id(event_id):
+            event = Event.objects.from_event_id(event_id, project.id)
+
+            if event is None:
+                return Response({'detail': 'Event not found'}, status=404)
+
+            event_id = event.event_id
+
+        snuba_event = SnubaEvent.get_event(project.id, event_id)
```

Comment:
Maybe this `get_event` should actually be part of the Manager too, like a version of `Event.objects.from_event_id` that returns SnubaEvents instead of old Events?

Reply:
Do you mean something like this? 70b3917